martha/martha_v3.js

```diff
@@ -347,12 +350,54 @@ async function retrieveFromServers(params) {
     let fileName;
     let localizationPath;
     let accessUrl;
+    let passports;
 
     // `fetch` below might time out in a way that we want to report as an error. Since the timeout
     // interrupts us while we're waiting for a response, we need to capture what we were about to
     // try doing just before we do it so that we can provide that detail in the error report.
     let hypotheticalErrorMessage;
 
+    const getAccessUrl = async (accessUrlAuth, httpsAccessUrl, accessToken, auth) => {
+        if (accessUrlAuth === AccessUrlAuth.PASSPORT) {
+            if (passports) {
+                try {
+                    return await apiAdapter.postJsonTo(httpsAccessUrl, null, {passports});
+                } catch (error) {
+                    console.log(`Passport authorized request failed for ${httpsAccessUrl} with error ${error}`);
+                }
+            }
```

Comment:
```suggestion
        if (accessUrlAuth === AccessUrlAuth.PASSPORT && passports) {
            try {
                return await apiAdapter.postJsonTo(httpsAccessUrl, null, {passports});
            } catch (error) {
                console.log(`Passport authorized request failed for ${httpsAccessUrl} with error ${error}`);
            }
```

Reply:
This is actually not the same because when there is no passport the final `else` statement at the end of the function will throw an error. But I am going to change this back to a switch.